packages/application-generic/src/usecases/trigger-multicast/trigger-multicast.usecase.ts

```diff
@@ -1,139 +1,207 @@
+        to: recipients,
+        actor,
+        userId,
+      } = command;
+
+      const mappedRecipients = Array.isArray(recipients)
+        ? recipients
+        : [recipients];
+
+      const { singleSubscribers, topicKeys } =
+        this.splitByRecipientType(mappedRecipients);
+
+      await this.sendToProcessSubscriberService(
+        command,
+        Array.from(singleSubscribers.values()),
+        SubscriberSourceEnum.SINGLE
+      );
+
+      const isEnabled = await this.getFeatureFlag.execute(
+        GetFeatureFlagCommand.create({
+          environmentId,
+          organizationId,
+          userId,
+          key: FeatureFlagsKeysEnum.IS_TOPIC_NOTIFICATION_ENABLED,
         })
       );
 
-      await this.validateSubscriberIdProperty(mappedRecipients);
+      if (!isEnabled) {
+        return;
+      }
 
-      const jobs = this.mapSubscribersToJobs(mappedRecipients, command);
+      const topics = await this.getTopicsByTopicKeys(
+        organizationId,
+        environmentId,
+        topicKeys
+      );
 
-      await this.subscriberProcessQueueAddBulk(jobs);
-    }
-  }
+      this.validateTopicExist(topics, topicKeys);
 
-  @CachedEntity({
-    builder: (command: { triggerIdentifier: string; environmentId: string }) =>
-      buildNotificationTemplateIdentifierKey({
-        _environmentId: command.environmentId,
-        templateIdentifier: command.triggerIdentifier,
-      }),
-  })
-  private async getNotificationTemplateByTriggerIdentifier(command: {
-    triggerIdentifier: string;
-    environmentId: string;
-  }) {
-    return await this.notificationTemplateRepository.findByTriggerIdentifier(
-      command.environmentId,
-      command.triggerIdentifier
-    );
-  }
+      const topicIds = topics.map((topic) => topic._id);
+      const singleSubscriberIds = Array.from(singleSubscribers.keys());
+      const subscriberFetchBatchSize = 500;
+      let subscribersList: ISubscribersDefine[] = [];
+      const getTopicDistinctSubscribersGenerator =
+        await this.topicSubscribersRepository.getTopicDistinctSubscribers({
+          query: {
+            _organizationId: organizationId,
+            _environmentId: environmentId,
+            topicIds: topicIds,
+            excludeSubscribers: singleSubscriberIds,
+          },
+          batchSize: subscriberFetchBatchSize,
+        });
 
-  @Instrument()
-  private async validateSubscriberIdProperty(
-    to: ISubscribersSource[]
-  ): Promise<boolean> {
-    for (const subscriber of to) {
-      const subscriberIdExists =
-        typeof subscriber === 'string' ? subscriber : subscriber.subscriberId;
-
-      if (Array.isArray(subscriberIdExists)) {
-        throw new ApiException(
-          'subscriberId under property to is type array, which is not allowed please make sure all subscribers ids are strings'
-        );
+      for await (const topicSubscriber of getTopicDistinctSubscribersGenerator) {
+        const externalSubscriberId = topicSubscriber._id;
```

Comment:
Actually, it is `_id` that we get after we execute aggregation `$group` it was not intuitive to me as well which is why i created this local const in order to be more informative.
Now that i am looking at it, maybe it will be a good idea to rename `topicSubscriber `